context/app/static/js/components/biomarkers/hooks.ts

```diff
@@ -0,0 +1,24 @@
+import { useCallback } from 'react';
+import { useGeneList } from '../genes/hooks';
+import { useBiomarkersSearchState } from './BiomarkersSearchContext';
+
+export function useResultsList() {
+  const { search } = useBiomarkersSearchState();
+  const { data, ...rest } = useGeneList(search);
+
+  const genesList = data?.flatMap((page) => page.genes) ?? [];
+
+  const hasMore = data?.[0]?.pagination?.page !== data?.[0]?.pagination?.total_pages;
+
+  // TODO: Once proteins are added to the API, we'll need to add them here as well
+
+  return { genesList, hasMore, ...rest };
+}
+
+export function useViewMore() {
+  const { size, setSize, isLoading, isValidating } = useResultsList();
+  return useCallback(async () => {
+    if (isLoading || isValidating) return;
+    await setSize(size + 1);
+  }, [isLoading, isValidating, setSize, size]);
+}
```

Comment:
Realistically, I should be using `useGenesList` here instead; both `useResultsList` and `useViewMore` are going to grow in complexity as we introduce proteins to this page, so having these hooks be decoupled from one another will be useful.

Reply:
I added a local `useCurrentGenesList` hook to avoid repeatedly needing to access the context to pull out the search value to provide to `useGenesList` which the `useResultsList` and `useViewMore` hooks can both use to access the genes SWR data.